src/main/java/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package strategy;
+
+import handler.OperationHandler;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public void handleOperation(FruitTransaction transaction) {
+        OperationHandler handler = handlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.handle(transaction);
+        } else {
+            throw new RuntimeException("No handler found for operation: "
+                    + transaction.getOperation());
+        }
```

Comment:
Avoid using if-else constructions for choosing operation handlers. It is recommended to use the Strategy pattern instead, which you've started to implement. Consider throwing the exception directly from the map's get method by using `Objects.requireNonNull` or `Optional`.